src/chk/chk_engine.c

```diff
@@ -135,22 +167,22 @@ chk_engine_exit(struct chk_instance *ins, uint32_t ins_status, uint32_t pool_sta
 }
 
 static uint32_t
-chk_engine_find_slowest(struct chk_instance *ins)
+chk_engine_find_slowest(struct chk_instance *ins, bool *done)
 {
-	uint32_t		 phase = CHK__CHECK_SCAN_PHASE__DSP_DONE;
-	uint32_t		 base = ins->ci_bk.cb_phase;
 	struct chk_pool_rec	*cpr;
+	uint32_t		 phase = CHK__CHECK_SCAN_PHASE__DSP_DONE;
+	int			 running = 0;
 
 	d_list_for_each_entry(cpr, &ins->ci_pool_list, cpr_link) {
-		if (cpr->cpr_phase <= base) {
-			phase = cpr->cpr_phase;
-			break;
-		}
-
-		if (cpr->cpr_phase < phase)
-			phase = cpr->cpr_phase;
+		if (cpr->cpr_bk.cb_phase < phase)
+			phase = cpr->cpr_bk.cb_phase;
+		if (!cpr->cpr_done)
+			running++;
 	}
 
+	if (done != NULL && running == 0)
+		*done = true;
```

Comment:
(phase == ..._DSP_DONE)?   then you do not need running?

Reply:
Here, the logic is that: if no pool scan is running "running == 0", then means the check on the engine is done.